### DISCUSSION TITLE
New Idea on Geometry Nodes Integration

### DISCUSSION THREAD

**Discussion by Clonkh** [hlorus/CAD_Sketcher#415] [Feature Requests]
Hello,

Would it be thinkable to create a CAD zone in Geometry nodes? Within that zone we could potentially get real CAD base on mathematically described volume sets that work well with intersect and combine operations. Only through the 'CAD Output' node that closes such a CAD zone the model would be converted to Vertex based mesh geometry. I could imagine working with a set of nodes like:

- Plane
- Sketch
- Extrude
- Spin
- etc

**Comment by hlorus** [comment]
Not really with the current python api. The node system is available only as a Ui element. Proper integration into existing node systems like shader nodes or geometry nodes is not possible afaik.

**Comment by Clonkh** [comment]
boo! Well we can't help it I guess. But such a true CAD environment would be awesome! Thx for the reply.
